pkg/monitor/worker.go

```diff
@@ -227,8 +232,36 @@ out:
 	log.Debug("stopping monitoring")
 }
 
+// Monitor represents a consistent interface for different monitoring components
+type Monitor interface {
+	Monitor(context.Context) []error
+}
+
+// noOpMonitor is a no operation monitor
+type noOpMonitor struct{}
+
+func (no noOpMonitor) Monitor(context.Context) []error {
+	return []error{}
+}
+
+func (mon *monitor) newNSGMonitor(log *logrus.Entry, oc *api.OpenShiftCluster, subscriptionID, tenantID string, e metrics.Emitter, dims map[string]string, wg *sync.WaitGroup) Monitor {
+	token, err := mon.env.FPNewClientCertificateCredential(tenantID)
+	if err != nil {
+		log.Error("Unable to create FP Authorizer for NSG monitoring.", err)
+		mon.clusterm.EmitGauge(nsg.MetricUnsuccessfulFPCreation, int64(1), dims)
```

Comment:
maybe make this metric more generic -> `failedNsgMonitorCreation` and use it below as well?  curious on your thoughts.